pkg/kubelet/stats/cri_stats_provider_windows.go

```diff
@@ -79,6 +81,73 @@ func (p *criStatsProvider) listContainerNetworkStats() (map[string]*statsapi.Net
 	return networkStats, nil
 }
 
+func (p *criStatsProvider) addCRIPodContainerStats(criSandboxStat *runtimeapi.PodSandboxStats,
+	ps *statsapi.PodStats, fsIDtoInfo map[runtimeapi.FilesystemIdentifier]*cadvisorapiv2.FsInfo,
+	containerMap map[string]*runtimeapi.Container,
+	podSandbox *runtimeapi.PodSandbox,
+	rootFsInfo *cadvisorapiv2.FsInfo,
+	updateCPUNanoCoreUsage bool) {
+	for _, criContainerStat := range criSandboxStat.Windows.Containers {
+		container, found := containerMap[criContainerStat.Attributes.Id]
+		if !found {
+			continue
+		}
+		// Fill available stats for full set of required pod stats
+		cs := p.makeWinContainerStats(criContainerStat, container, rootFsInfo, fsIDtoInfo, podSandbox.GetMetadata())
```

Comment:
To make sure my understanding of your comment is correct, you are suggesting we get the usage from the cache if the stats.Cpu.UsageNanoCores value is nil, something of the sort:

```
if stats.Cpu.UsageNanoCores != nil {
			result.CPU.UsageNanoCores = &stats.Cpu.UsageNanoCores.Value
		} else {
			// If UsageNanoCores is nil, retrieve the value from the cache
			result.CPU.UsageNanoCores = p.getContainerUsageNanoCores(stats)
		}
```

Reply:
@claudiubelu here is the comment from @jsturtevant from a previous iteration: https://github.com/kubernetes/kubernetes/pull/116968#discussion_r1157443250

Reply:
I looked into this a bit more and since [containerd](https://github.com/containerd/containerd/blob/8a6c8a96c0de336b15cbdc4693605add6868c264/pkg/cri/server/container_stats_list.go#L82) and kubelet are keeping track of this metric.  In the case of kubelet it is only so it can calculate `UsageNanoCores` over the time period from what I can tell.

In the CRI only scenario, this should be used from containerd since it is being calculated by the runtime. @haircommander @bobbypage thoughts?

I don't think this should hold up this PR for windows, unless it needs to be tracked by kubelet.

Reply:
yeah the CRI is calculating it now so it can have a collection frequency that's independent of the kubelet's